### PR TITLE
Add media player support for mps-youtube

### PR DESCRIPTION
**Pull request by xobust**
I'm using mpsyt so I thought that it would be nice to have it working with Snip.
It works by taking the track name from the window title like VLC. 
I had some trouble implementing hotkeys but they work okay now.

### CODE REVIEW

**Comment by dlrudie** [comment]
Thanks for contributing! I'll comment throughout where I feel things should be corrected before I commit.

File: Snip/Snip.cs [hunk 1]
```diff
@@ -161,6 +161,9 @@ private void LoadSettings()
                 case Globals.MediaPlayerSelection.quodlibet:
                     this.ToggleQuodlibet();
                     break;
```
Comment (dlrudie) [inline Snip/Snip.cs]: Add a blank line after this.

File: Snip/Players/mpsyt.cs [hunk 2]
```diff
@@ -1,4 +1,24 @@
-﻿namespace Winter
+﻿#region File Information
+/*
+ * Copyright (C) 2012-2016 David Rudie
```
Comment (dlrudie) [inline Snip/Players/mpsyt.cs]: You can change this to just 2017.

File: Snip/Globals.cs [hunk 3]
```diff
@@ -34,6 +34,9 @@ public static class Globals
         public const string NewLineVariable = "$$n";
         public const string TrackIdVariable = "$$i";
 
+        // Delay when using WM_Char in ms
```
Comment (dlrudie) [inline Snip/Globals.cs]: Why is a delay necessary?

File: Snip/Players/mpsyt.cs [hunk 4]
```diff
@@ -65,37 +61,39 @@ public override void Unload()
 
         public override void ChangeToNextTrack()
         {
```
Comment (dlrudie) [inline Snip/Players/mpsyt.cs]: All of the keys listed here can be added to the Snip.cs file under ToggleHotkeys() and KeyboardHook_KeyPressed() methods.

File: README.md [hunk 5]
```diff
@@ -8,7 +8,7 @@ ABOUT
 This is a small application that sits in the system tray and updates a text
 file with the currently playing audio track.
 
-It supports Spotify, iTunes, Winamp, foobar2000, and VLC.
+It supports Spotify, iTunes, Winamp, foobar2000, VLC and mps-youtube.
```
Comment (dlrudie) [inline README.md]: I prefer Chicago style of writing and like a comma in my list.  VLC, and mps-youtube.

File: README.md [hunk 6]
```diff
@@ -80,6 +81,14 @@ determine if VLC is running.  If the title bar is anything different then it
 will not be found.  If a track is currently playing when you start Snip you
 just have to push stop and Snip will find VLC.
 
+Mps-youtube
+===
```
Comment (dlrudie) [inline README.md]: ===========

File: README.md [hunk 7]
```diff
@@ -80,6 +81,14 @@ determine if VLC is running.  If the title bar is anything different then it
...
+Mpsyt needs to be started using directly using the .exe file witch is located in your the
```
Comment (dlrudie) [inline README.md]: This paragraph needs corrected. I'll work on this later.

File: README.md [hunk 8]
```diff
@@ -93,4 +102,4 @@ HOTKEYS
 
 Note
 ----
-Not all hotkeys work between Spotify, iTunes, Winamp, foobar2000, and VLC.
+Not all hotkeys work between Spotify, iTunes, Winamp, foobar2000,mps-youtube and VLC.
```
Comment (dlrudie) [inline README.md]: Missing a space.
"Not all hotkeys work between Spotify, iTunes, Winamp, foobar2000, VLC, and mps-youtube.

File: Snip/Players/mpsyt.cs [hunk 9]
```diff
@@ -29,8 +29,27 @@ public override void Update()
...
-                }else
+                    // Remove the '- mpsyt' part of the window title to recive the song title
+                    int lastHyphen = windowTitle.LastIndexOf("-", StringComparison.OrdinalIgnoreCase);
+
+                    //If we find a hyphen a track is playing there is no track playing
+                    if (lastHyphen > 0)
+                    {
+                        string songTitle = windowTitle.Substring(0, lastHyphen).Trim();
+
+                        if (Globals.SaveAlbumArtwork)
```
Comment (dlrudie) [inline Snip/Players/mpsyt.cs]: This is not necessary as artwork is not supported.

File: Snip/Players/mpsyt.cs [hunk 10]
```diff
@@ -29,8 +29,27 @@ public override void Update()
                 // if mpsyt was ran from console there will exist 
                 if(windowTitle.Length > 0)
                 {
-
-                }else
+                    // Remove the '- mpsyt' part of the window title to recive the song title
```
Comment (dlrudie) [inline Snip/Players/mpsyt.cs]: Remove the '- mpsyt' part of the window title to receive the song title

File: Snip/Players/mpsyt.cs [hunk 11]
```diff
@@ -29,8 +29,27 @@ public override void Update()
...
+                    int lastHyphen = windowTitle.LastIndexOf("-", StringComparison.OrdinalIgnoreCase);
+
+                    //If we find a hyphen a track is playing there is no track playing
```
Comment (dlrudie) [inline Snip/Players/mpsyt.cs]: // If a hyphen exists in the titlebar we can assume a track is playing.

File: Snip/Snip.cs [hunk 12]
```diff
@@ -234,6 +234,9 @@ private void PlayerSelectionCheck(object sender, EventArgs e)
             else if (sender == this.toolStripMenuItemVlc)
             {
                 this.ToggleVLC();
+            }else if (sender == this.toolStripMenuItemMpsyt)
```
Comment (dlrudie) [inline Snip/Snip.cs]: }
else if...

File: Snip/Snip.cs [hunk 13]
```diff
@@ -317,6 +324,24 @@ private void ToggleVLC()
...
+
+            Globals.CurrentPlayer.Unload();
+            Globals.CurrentPlayer = new mpsyt();
+            Globals.CurrentPlayer.Load();
+
+            Globals.PlayerSelection = Globals.MediaPlayerSelection.mpsyt;
+            TextHandler.UpdateTextAndEmptyFilesMaybe(Globals.ResourceManager.GetString("SwitchedTompsyt"));
+        }
+
+
```
Comment (dlrudie) [inline Snip/Snip.cs]: Extra blank line here.

File: Snip/Players/mpsyt.cs [hunk 14]
```diff
@@ -25,7 +25,16 @@ public override void Update()
                     windowTitle = process.MainWindowTitle;
                 }
 
-                System.Console.WriteLine(windowTitle);
+                // We need to check if the title exists,
+                // if mpsyt was ran from console there will exist
```
Comment (dlrudie) [inline Snip/Players/mpsyt.cs]: I don't understand this sentence. :(

File: Snip/Players/mpsyt.cs [hunk 15]
```diff
@@ -25,7 +25,16 @@ public override void Update()
                     windowTitle = process.MainWindowTitle;
                 }
 
-                System.Console.WriteLine(windowTitle);
+                // We need to check if the title exists,
+                // if mpsyt was ran from console there will exist 
+                if(windowTitle.Length > 0)
```
Comment (dlrudie) [inline Snip/Players/mpsyt.cs]: if (windowTitle.Length > 0)

File: Snip/Players/mpsyt.cs [hunk 16]
```diff
@@ -25,7 +25,16 @@ public override void Update()
...
+                {
+
+                }else
```
Comment (dlrudie) [inline Snip/Players/mpsyt.cs]: }
else
{

File: Snip/Resources/Strings.txt [hunk 17]
```diff
@@ -41,6 +41,10 @@ mpsytIsNotRunning=mps-youtube is not currently running
 ; selected media file or when the user stops/pauses a playing track.
 NoTrackPlaying=No Track Playing
 
+; This text is saved to the Snip.txt file when mpsyt apears to have no title
```
Comment (dlrudie) [inline Snip/Resources/Strings.txt]: This is really unnecessary. The instructions should be clear in README.md.
Just display NoTrackPlaying instead.

File: Snip/Globals.cs (hunk 3 above)
Comment (xobust) [inline Snip/Globals.cs]: It is strange I think that the window does not accept any input when holding ctrl and alt that is why I had to add a delay. I'm not a winapi export but there might be a way to signal that ctrl and alt are no longer pressed.

**Comment by xobust** [comment]
I corrected some of the things you pointed out. Writing documentation might not be my strong suit :P 
Please tell me if you need any more help 👍

**Comment by xobust** [comment]
@dlrudie What more needs fixing?

**Comment by dlrudie** [comment]
I'm going to get back to this soon. I may have time this weekend to look into it.

**Pull request closed without merging**